### DISCUSSION TITLE
Multiple yarn.lock files

### DISCUSSION THREAD

**Discussion by KnisterPeter** [renovatebot/renovate#9212] [Request Help] [answered]
Hi there,

we have a bigger monorepo (roughly 160 packages).
Some of the packages could be seen as 'side-topics' and could/should be build/managed separatly.
While this is possible with yarn workspaces to have multiple `yarn.lock` files in a repository, I'm not sure if this is a supported pattern for renovate. How to configure this?

**Comment by rarkins** [comment, marked as answer]
I'm not aware of a reason why it wouldn't work, although I wouldn't bet my life on it.

Renovate will first locate and parse all `package.json` files. In a typical workspaces project, most of those won't have their own `yarn.lock` - only the ones at the root of a workspace.

Then, on a second round through it will determine which of those `package.json` are the root of workspaces, parse the `packages` globs, and work out which sub-packages apply, and associate them with the root package yarn.lock. Whenever any of those sub-packages need updating, we'll run `yarn` in the workspace directory to update the workspaces lock file.

If any packages are excluded from the workspaces packages globs and have their own yarn.lock, they should be updated independently.

**Reply by KnisterPeter** [reply]
Thank you. I will give it a try

**Comment by KnisterPeter** [comment]
It does work as expected. Thank you. :tada:
